File: Python/shared/client_side_extension_item.py
# Exact full port of ClientSideExtensionItem.cs

import xml.etree.ElementTree as ET
from .baseline_item import BaselineItem
from .effective_state_sources import EffectiveStateSources

class ClientSideExtensionItem(BaselineItem):
    def __init__(self, node: ET.Element):
        super().__init__()
        self.m_sCSEType = node.tag  # LocalName equivalent in ElementTree
        guid_elem = node.find("GUID")
        name_elem = node.find("Name")
        self.m_sGuid = guid_elem.text.strip() if guid_elem is not None and guid_elem.text else ""
        self.m_sName = name_elem.text.strip() if name_elem is not None and name_elem.text else ""

    def SortKey(self) -> str:
        return (self.__class__.__name__ + "!" + self.m_sCSEType + "!" + self.m_sGuid).lower()

    def Evaluate(self, sources: EffectiveStateSources, x_output_doc: ET.Element):
        root = x_output_doc.getroot()
        if root is None:
            root = ET.SubElement(x_output_doc, "root")
        elem = ET.SubElement(root, self.m_sCSEType)
        ET.SubElement(elem, "GUID").text = self.m_sGuid
        ET.SubElement(elem, "Name").text = self.m_sName
        root.append(elem)

File: CS/PolicyAnalyzer/SharedCode/ClientSideExtensionItem.cs
using System;
using System.Xml;

namespace SharedCode
{
	// Token: 0x02000049 RID: 73
	internal class ClientSideExtensionItem : BaselineItem
	{
		// Token: 0x060000F1 RID: 241 RVA: 0x00002F2B File Offset: 0x0000112B
		public ClientSideExtensionItem(XmlNode node)
		{
			this.m_sCSEType = node.LocalName;
			this.m_sGuid = node["GUID"].InnerText;
			this.m_sName = node["Name"].InnerText;
		}

		// Token: 0x060000F2 RID: 242 RVA: 0x0000BBBC File Offset: 0x00009DBC
		public override string SortKey()
		{
			return string.Concat(new string[]
			{
				base.GetType().Name,
				"!",
				this.m_sCSEType,
				"!",
				this.m_sGuid
			}).ToLower();
		}

		// Token: 0x060000F3 RID: 243 RVA: 0x0000BC0C File Offset: 0x00009E0C
		public override void Evaluate(EffectiveStateSources sources, XmlDocument xOutputDoc)
		{
			XmlElement xmlElement = xOutputDoc.CreateElement(this.m_sCSEType);
			XmlElement xmlElement2 = xOutputDoc.CreateElement("GUID");
			XmlElement xmlElement3 = xOutputDoc.CreateElement("Name");
			xmlElement2.InnerText = this.m_sGuid;
			xmlElement3.InnerText = this.m_sName;
			xmlElement.AppendChild(xmlElement2);
			xmlElement.AppendChild(xmlElement3);
			xOutputDoc.DocumentElement.AppendChild(xmlElement);
		}

		// Token: 0x04000698 RID: 1688
		private string m_sCSEType;

		// Token: 0x04000699 RID: 1689
		private string m_sGuid;

		// Token: 0x0400069A RID: 1690
		private string m_sName;
	}
}
